lib/galaxy/jobs/__init__.py

```diff
@@ -1743,7 +1743,14 @@ def _change_ownership( self, username, gid ):
         job = self.get_job()
         # FIXME: hardcoded path
         external_chown_script = self.get_destination_configuration("external_chown_script", None)
-        cmd = [ '/usr/bin/sudo', '-E', external_chown_script, self.working_directory, username, str( gid ) ]
+
+        cmd = [ '/usr/bin/sudo', '-E', ]
+        for v in ["PATH", "LD_LIBRARY_PATH", "PKG_CONFIG_PATH"]:
```

Comment:
Well, I don't think forcing users to build special version of the python interpreter just for this purpose is great. I kind of like injecting PATH (and LD_LIBRARY_PATH -urgs-, if necessary) instead of propagating all variables with `sudo -E`.